src/CribblyBackend.DataAccess/Repositories/TeamRepository.cs

```diff
@@ -63,9 +63,28 @@ public async Task<int> Create(Team team)
             return (await _connection.QueryAsync<int>(@"SELECT LAST_INSERT_ID()")).First();
         }
 
-        public void Delete(Team team)
+        public async Task Delete(Team team)
         {
-            throw new System.NotImplementedException();
+            if (this.GetById(team.Id).Result == null)
+            {
+                throw new System.ArgumentNullException("Team not found, nothing to delete");
+            };
+
+            await _connection.ExecuteAsync(
+                @"
+                    SET foreign_key_checks = 0;
```

Comment:
Is this a good strategy in production?

Reply:
I have yet to find a better way to do it. I will keep digging

Reply:
Maybe using a transaction would help?

Reply:
Probably the thing you'd have to do is reverse the order -- set the ID to null on the player, then delete the team.

Reply:
I figured out my issue. I needed to set the TeamId to null instead of 0. 🤦